command/agent/auth/aws/aws.go

```diff
@@ -238,7 +238,7 @@ func (a *awsMethod) Authenticate(ctx context.Context, client *api.Client) (retTo
 		defer a.credLock.Unlock()
 
 		var err error
-		data, err = awsauth.GenerateLoginData(a.lastCreds, a.headerValue)
+		data, err = awsauth.GenerateLoginData(a.lastCreds, a.headerValue, "")
```

Comment:
Thoughts on having this read from the AWS_REGION environment variable or otherwise be configurable?

Reply:
I was _thinking_ about it. There's code [here](https://github.com/hashicorp/vault/blob/master/vault/seal/awskms/awskms.go#L91-L117) that could be moved into the `helper/awsutil` library and reused. Then I worried that it might result in unexpected new behavior because of how sometimes the `AWS_DEFAULT_REGION` is automatically set, but it's been defaulting to `us-east-1`, so that would change for people running in environments with that variable. I ended up stripping out code to grab the env var there, but I'm flexible on that approach. Also maybe with the changes you've suggested that will become a non-issue, I haven't hacked at them yet.

Reply:
Hmm, good point. I wonder how many people run the agent using the `AWS[_DEFAULT]_REGION` environment variable? Most system services don't load login profiles and so people have to go out of their way to set these variables in services.

As is, there's no way to specify an alternate region in here. Is there a reason for the agent to talk to AWS outside of this? If not, I think it might make sense to just document this as a potentially breaking change and people can unset it if they want to keep the old behavior.

Reply:
I can't find any statistics on how people tend to run on AWS. At two of the companies I've been at, it was very EC2-centric, so I do admittedly tend to think of it that way. 

I will PR the code I was thinking of into this branch and @ you on it and we can decide whether to do it.

Reply:
This was added via https://github.com/hashicorp/vault/pull/6284